### DISCUSSION TITLE
How to monitor logs

### DISCUSSION THREAD

**Discussion by UlricQin** [ccfos/nightingale#1254] [Show and tell]
两个方案：

1、在产生日志的机器上，部署categraf，流式读取日志，根据正则提取指标，比如统计Error关键字出现的次数，统计接口的成功率、延迟等，具体参考：[从应用日志中提取监控metrics](https://flashcat.cloud/blog/log-monitoring-mtail/)

2、把日志收集到中心，比如放到ElasticSearch中，然后在夜莺里配置告警规则，具体可以参考：[ElasticSearch日志告警](https://flashcat.cloud/blog/elasticsearch-alert/)

**Comment by jiaomingliang** [comment]
第一种方案需要购买企业版吗

**Reply by UlricQin** [reply]
不需要，开源版本就支持

**Reply by jujunchen** [reply]
第一种好像要自己定义告警规则

**Comment by HAHA-LIU** [comment]
作者好，我有一个问题如下：

1. 看了 "从应用日志中提取监控metrics"，有点没太理解具体怎么用。categraf 集成了mtail，修改了配置文件，然后具体页面怎么操作，不是很会，可否出一期日志监控的教程。

categraf 修改了配置文件：
![image](https://user-images.githubusercontent.com/50697108/208051542-30d4e47a-5e7c-4046-bed1-5835f3b6e4f7.png)

**Reply by kongfei605** [reply]
prog是mtail的规则文件目录，不是日志目录。插件地址 https://github.com/flashcatcloud/categraf/tree/main/inputs/mtail ，这个地址社区反馈和细节会更多一些
